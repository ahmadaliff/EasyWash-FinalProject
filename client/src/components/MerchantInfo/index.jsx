import PropTypes from 'prop-types';
import { useState } from 'react';

import classes from '@components/MerchantInfo/style.module.scss';
import { Avatar, Skeleton } from '@mui/material';
import { FormattedMessage } from 'react-intl';
import config from '@config/index';

const MerchantInfo = ({ merchant }) => {
  const [loading, setLoading] = useState(true);
  return (
    <div className={classes.merchant}>
      <div className={classes.imgWrap}>
        {merchant?.imagePath ? (
          <>
            {loading && <Skeleton variant="square" className={classes.skeleton} />}
            <img
              src={`${config.api.server}${merchant?.imagePath}`}
              alt={merchant?.name}
              loading="lazy"
              onLoad={() => setLoading(false)}
            />
          </>
        ) : (
          <Avatar className={classes.avatar} variant="square">
            {merchant?.name?.split(' ')[0][0]}
            {merchant?.name?.split(' ').length > 1 && merchant?.name?.split(' ')[1][0]}
          </Avatar>
        )}
      </div>
      <div>
        <h4 className={classes.merchantInfoHeader}>
          {/* <FormattedMessage id="app_merchant_header" /> */}
          {merchant?.name}
        </h4>
        <span className={classes.profileInfo}>
          <p className={classes.profileLabel}>
            <FormattedMessage id="app_description" />
          </p>
          <p className={classes.profileText}>: {merchant?.description}</p>
        </span>
      </div>
    </div>
  );
};

MerchantInfo.propTypes = {
  merchant: PropTypes.object,
};

export default MerchantInfo;
