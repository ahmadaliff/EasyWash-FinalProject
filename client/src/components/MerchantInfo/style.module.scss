@import '@styles/base';
.merchant {
  --image-size-sm: 5rem;
  --image-size-md: 5rem;
  @include directionY;
  @include apply-at-sm-size {
    @include directionX;
  }
  gap: 1rem;
  position: relative;
  .imgWrap {
    align-self: center;
    @include apply-at-md-size {
      align-self: auto;
    }
    height: var(--image-size-sm);
    width: var(--image-size-sm);
    @include apply-at-md-size {
      height: var(--image-size-md);
      width: var(--image-size-md);
    }
    overflow: hidden;
    position: relative;
    img {
      object-fit: cover;
      object-position: top;
      height: var(--image-size-sm);
      width: var(--image-size-sm);
      @include apply-at-md-size {
        height: var(--image-size-md);
        width: var(--image-size-md);
      }
    }
    .skeleton {
      position: absolute;
      height: var(--image-size-sm);
      width: var(--image-size-sm);
      @include apply-at-md-size {
        height: var(--image-size-md);
        width: var(--image-size-md);
      }
    }
    .avatar {
      background-color: var(--color-primary);
      height: var(--image-size-sm);
      width: var(--image-size-sm);
      @include apply-at-md-size {
        height: var(--image-size-md);
        width: var(--image-size-md);
      }
    }
    .buttonEdit {
      position: absolute;
      display: none;
      top: 50%;
      left: 50%;
      transform: translate(-50%, -50%);
    }
  }
  .merchantInfoHeader {
    margin: 0;
    font-size: medium;
  }
  .profileInfo {
    @include directionX;
    font-size: 0.75rem;
    font-weight: 500;
    .profileLabel {
      float: left;
      width: 8rem;
      margin: 0;
    }
    .profileText {
      margin: 0;
      width: 100%;
      max-width: 10rem;
      @include apply-at-sm-size {
        max-width: 15rem;
      }
    }
  }
  .chatButton {
    position: absolute;
    top: 0;
    right: 0;
  }
}
