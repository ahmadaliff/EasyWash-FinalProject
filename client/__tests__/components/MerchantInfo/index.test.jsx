import MerchantInfo from '@components/MerchantInfo';
import { render } from '@utils/testHelper';

let wrapper;

jest.mock('react-router-dom', () => ({
  useNavigate: () => jest.fn(),
}));

const mockProps = {
  merchant: {
    name: 'merchant',
    imagePath: null,
    distance: 5,
    Favorits: [],
    userId: 2,
  },
};

beforeEach(() => {
  wrapper = render(<MerchantInfo {...mockProps} />);
});
describe('MerchantInf0 Component', () => {
  test('Correct render', () => {
    const { getByTestId } = wrapper;
    expect(getByTestId('merchant-info')).toBeInTheDocument();
  });
  test('Should match with snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
