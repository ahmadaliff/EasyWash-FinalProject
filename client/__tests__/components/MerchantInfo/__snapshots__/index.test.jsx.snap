// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MerchantInf0 Component Should match with snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="merchant"
        data-testid="merchant-info"
      >
        <div
          class="imgWrap"
        >
          <div
            class="MuiAvatar-root MuiAvatar-square MuiAvatar-colorDefault avatar css-18gxrea-MuiAvatar-root"
          >
            m
          </div>
        </div>
        <div>
          <h4
            class="merchantInfoHeader"
          >
            merchant
          </h4>
          <span
            class="profileInfo"
          >
            <p
              class="profileText"
            />
          </span>
        </div>
        <button
          aria-label="Pesan"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeMedium chatButton css-zylse7-MuiButtonBase-root-MuiIconButton-root"
          data-mui-internal-clone-element="true"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="AddCommentOutlinedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M22 4c0-1.1-.9-2-2-2H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h14l4 4V4zm-2 13.17L18.83 16H4V4h16v13.17zM13 5h-2v4H7v2h4v4h2v-4h4V9h-4z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="merchant"
      data-testid="merchant-info"
    >
      <div
        class="imgWrap"
      >
        <div
          class="MuiAvatar-root MuiAvatar-square MuiAvatar-colorDefault avatar css-18gxrea-MuiAvatar-root"
        >
          m
        </div>
      </div>
      <div>
        <h4
          class="merchantInfoHeader"
        >
          merchant
        </h4>
        <span
          class="profileInfo"
        >
          <p
            class="profileText"
          />
        </span>
      </div>
      <button
        aria-label="Pesan"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeMedium chatButton css-zylse7-MuiButtonBase-root-MuiIconButton-root"
        data-mui-internal-clone-element="true"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="AddCommentOutlinedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M22 4c0-1.1-.9-2-2-2H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h14l4 4V4zm-2 13.17L18.83 16H4V4h16v13.17zM13 5h-2v4H7v2h4v4h2v-4h4V9h-4z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
